src/components/base/button.vue

```diff
@@ -0,0 +1,34 @@
+<template>
+	<button
+		:type="props.type"
+		:class="[
+			props.color === 'primary'
+				? 'bg-primary-700 text-white hover:bg-primary-800 focus:ring-primary-300 dark:bg-primary-600 dark:hover:bg-primary-700 dark:focus:ring-primary-800'
+				: '',
+			props.color === 'secondary'
+				? 'bg-primary-500 text-white hover:bg-primary-600 focus:ring-primary-100 dark:bg-primary-400 dark:hover:bg-primary-500 dark:focus:ring-primary-600'
+				: '',
+		]"
+		class="w-full rounded-lg px-5 py-3 text-center text-base font-medium focus:ring-4 sm:w-auto"
+		@click="emit('click')"
+	>
+		<slot />
+	</button>
+</template>
+
+<script setup lang="ts">
+import type { PropType } from 'vue';
+
+const props = defineProps({
+	type: {
+		type: String as PropType<'submit' | 'reset' | 'button'>,
+		required: true,
+	},
+	color: {
+		type: String as PropType<'primary' | 'secondary'>,
```

Comment:
Would this be a good time for Enums? Then we could also remove the "magic strings" that we use around the component? Making it more type-safe, less likely to type a string wrong.

Like this?

```vue
<template>
	<button
		:type="props.type"
		:class="[
			props.color === ButtonColor.Primary
				? 'bg-primary-700 text-white hover:bg-primary-800 focus:ring-primary-300 dark:bg-primary-600 dark:hover:bg-primary-700 dark:focus:ring-primary-800'
				: '',
			props.color === ButtonColor.Secondary
				? 'bg-primary-500 text-white hover:bg-primary-600 focus:ring-primary-100 dark:bg-primary-400 dark:hover:bg-primary-500 dark:focus:ring-primary-600'
				: '',
		]"
		class="w-full rounded-lg px-5 py-3 text-center text-base font-medium focus:ring-4 sm:w-auto"
		@click="emit('click')"
	>
		<slot />
	</button>
</template>

<script lang="ts">
export enum ButtonColor {
	Primary = 'primary',
	Secondary = 'secondary',
}
</script>

<script setup lang="ts">
import type { PropType } from 'vue';

const props = defineProps({
	type: {
		type: String as PropType<'submit' | 'reset' | 'button'>,
		required: true,
	},
	color: {
		type: String as PropType<ButtonColor>,
		required: true,
	},
});

const emit = defineEmits(['click']);
</script>
```
![image](https://github.com/OurUncomplicatedTemplates/astro-vue-tailwind-jwt-template/assets/53931453/9b328673-9a3b-4c1b-8efe-776349836eac)